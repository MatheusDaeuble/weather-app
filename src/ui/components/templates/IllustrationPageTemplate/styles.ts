import styled from 'styled-components/native';
import Typography from 'ui/components/atoms/Typography';
import { Spacing } from 'ui/theme';
import { colors } from 'ui/theme/colors';

export const Container = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  background: ${colors.white};
  padding: ${Spacing.SCALE_24}px;
  align-items: center;
`;

export const Content = styled.View`
  flex: 1;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

export const Title = styled(Typography).attrs({
  variant: 'h2',
  align: 'center',
})`
  margin-vertical: ${Spacing.SCALE_20}px;
`;

export const Description = styled(Typography).attrs({
  variant: 'body1',
  color: 'gray',
  align: 'center',
})``;
