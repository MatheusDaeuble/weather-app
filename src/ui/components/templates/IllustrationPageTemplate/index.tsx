import React from 'react';
import SafeArea from 'ui/containers/SafeArea';
import Illustration from 'ui/components/atoms/Illustration';
import { WINDOW_WIDTH } from 'util/dimensions';
import Button, { ButtonProps } from 'ui/components/atoms/Button';
import { IllustrationName } from 'ui/components/atoms/Illustration/types';
import { Container, Title, Description, Content } from './styles';

interface IllustrationPageTemplateProps {
  title?: string;
  description?: string;
  illustration?: IllustrationName;
  button?: ButtonProps;
}

const IllustrationPageTemplate = ({
  title = '',
  description = '',
  illustration,
  button,
}: IllustrationPageTemplateProps) => {
  return (
    <SafeArea>
      <Container>
        <Content>
          {!!illustration && (
            <Illustration name={illustration} width={WINDOW_WIDTH * 0.6} />
          )}
          <Title>{title}</Title>
          <Description>{description}</Description>
        </Content>
        {!!button && <Button {...button} />}
      </Container>
    </SafeArea>
  );
};

export default IllustrationPageTemplate;
