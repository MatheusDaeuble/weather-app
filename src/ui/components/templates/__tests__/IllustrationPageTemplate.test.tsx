import React from 'react';
import renderer from 'react-test-renderer';
import IllustrationPageTemplate from '../IllustrationPageTemplate';

describe('IllustrationPageTemplate', () => {
  it('Default', () => {
    const component = renderer
      .create(
        <IllustrationPageTemplate
          title='Test'
          description='testing'
          illustration='location'
          button={{ label: 'Test' }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Without title', () => {
    const component = renderer
      .create(
        <IllustrationPageTemplate
          description='testing'
          illustration='location'
          button={{ label: 'Test' }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Without description', () => {
    const component = renderer
      .create(
        <IllustrationPageTemplate
          title='Test'
          illustration='location'
          button={{ label: 'Test' }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Without illustration', () => {
    const component = renderer
      .create(
        <IllustrationPageTemplate
          title='Test'
          description='testing'
          button={{ label: 'Test' }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Without button', () => {
    const component = renderer
      .create(
        <IllustrationPageTemplate
          title='Test'
          description='testing'
          illustration='location'
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
