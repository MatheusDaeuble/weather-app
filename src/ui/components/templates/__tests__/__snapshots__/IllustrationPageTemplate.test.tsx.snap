// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IllustrationPageTemplate Default 1`] = `
<RCTSafeAreaView
  color="white"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingBottom": 48,
          "paddingLeft": 48,
          "paddingRight": 48,
          "paddingTop": 48,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "height": undefined,
            "width": 450,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M67.4 26.49s-33.146 16.903-40.16 54c-7.013 37.097 12.82 74.6 28.044 105.333 15.223 30.734 46.446 45.2 75.496 35.917 29.05-9.283 23.724-35.74 47.307-62.56 23.583-26.82 34.333-30.483 38.043-65.483 3.71-35-31.746-70.334-76.666-76.487-44.92-6.153-72.064 9.28-72.064 9.28z"
              fill={4281240031}
              opacity={0.08}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M107.363 128.572a1.408 1.408 0 01-.87 1.308 1.419 1.419 0 01-1.541-.304 1.42 1.42 0 01-.308-1.541c.107-.258.288-.479.52-.635a1.418 1.418 0 011.784.175c.265.264.414.623.415.997z"
              fill={4292927712}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M139.856 160.388a44.426 44.426 0 01-63.776-37.293 44.424 44.424 0 0111.108-32.142 44.433 44.433 0 0177.595 26.817 44.426 44.426 0 01-11.113 32.142 44.431 44.431 0 01-13.814 10.476z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeMiterlimit",
                ]
              }
              stroke={4280693304}
              strokeMiterlimit={10}
              strokeWidth={0.5}
            />
            <RNSVGPath
              d="M134.747 100.275c-3.957-5.444-9.71-7.98-15.714-7.98 0 0-14.703 0-19.056 13.16-1.604 4.85-1.25 10.12.696 14.843a27.235 27.235 0 002.75 5.107c3.794 5.483 15.61 20.456 15.61 20.456h2.57c3.297-4.58 11.6-19.796 14.087-24.666 3.12-6.134 3.6-14.67-.943-20.92zm-13.26 24.333a12.567 12.567 0 01-11.607-7.751 12.563 12.563 0 012.721-13.689 12.553 12.553 0 0119.325 1.902 12.552 12.552 0 012.117 6.978 12.553 12.553 0 01-3.676 8.892 12.558 12.558 0 01-8.89 3.681l.01-.013z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M137.323 100.545c-3.956-5.443-9.71-7.973-15.713-7.973 0 0-14.703 0-19.057 13.163-1.606 4.85-1.25 10.117.697 14.84a27.046 27.046 0 002.747 5.107c3.796 5.483 15.613 20.46 15.613 20.46s13.667-18.78 16.667-24.667c3.11-6.143 3.59-14.68-.954-20.93zm-15.83 20.404a8.61 8.61 0 01-8.454-10.297 8.614 8.614 0 0111.75-6.28 8.61 8.61 0 015.318 7.963 8.555 8.555 0 01-8.614 8.614z"
              fill={4285637118}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M121.446 103.522a8.694 8.694 0 00-8.193 5.79 7.447 7.447 0 0112.147-2.109 7.45 7.45 0 11-11.237 9.752 8.693 8.693 0 107.283-13.433z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        align="center"
        color="gray-dark"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#4B4F58",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 52,
              "lineHeight": 78,
              "textAlign": "center",
            },
            Object {
              "marginVertical": 40,
            },
          ]
        }
        variant="h2"
      >
        Test
      </Text>
      <Text
        align="center"
        color="gray"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#949494",
              "fontFamily": "OpenSans_400Regular",
              "fontSize": 28,
              "lineHeight": 42,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="body1"
      >
        testing
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E89DF",
          "borderRadius": 16,
          "justifyContent": "center",
          "minHeight": 54,
          "opacity": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
          "width": "100%",
        }
      }
    >
      <Text
        align="center"
        color="white"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 36,
              "lineHeight": 54,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="h3"
      >
        Test
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`IllustrationPageTemplate Without button 1`] = `
<RCTSafeAreaView
  color="white"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingBottom": 48,
          "paddingLeft": 48,
          "paddingRight": 48,
          "paddingTop": 48,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "height": undefined,
            "width": 450,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M67.4 26.49s-33.146 16.903-40.16 54c-7.013 37.097 12.82 74.6 28.044 105.333 15.223 30.734 46.446 45.2 75.496 35.917 29.05-9.283 23.724-35.74 47.307-62.56 23.583-26.82 34.333-30.483 38.043-65.483 3.71-35-31.746-70.334-76.666-76.487-44.92-6.153-72.064 9.28-72.064 9.28z"
              fill={4281240031}
              opacity={0.08}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M107.363 128.572a1.408 1.408 0 01-.87 1.308 1.419 1.419 0 01-1.541-.304 1.42 1.42 0 01-.308-1.541c.107-.258.288-.479.52-.635a1.418 1.418 0 011.784.175c.265.264.414.623.415.997z"
              fill={4292927712}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M139.856 160.388a44.426 44.426 0 01-63.776-37.293 44.424 44.424 0 0111.108-32.142 44.433 44.433 0 0177.595 26.817 44.426 44.426 0 01-11.113 32.142 44.431 44.431 0 01-13.814 10.476z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeMiterlimit",
                ]
              }
              stroke={4280693304}
              strokeMiterlimit={10}
              strokeWidth={0.5}
            />
            <RNSVGPath
              d="M134.747 100.275c-3.957-5.444-9.71-7.98-15.714-7.98 0 0-14.703 0-19.056 13.16-1.604 4.85-1.25 10.12.696 14.843a27.235 27.235 0 002.75 5.107c3.794 5.483 15.61 20.456 15.61 20.456h2.57c3.297-4.58 11.6-19.796 14.087-24.666 3.12-6.134 3.6-14.67-.943-20.92zm-13.26 24.333a12.567 12.567 0 01-11.607-7.751 12.563 12.563 0 012.721-13.689 12.553 12.553 0 0119.325 1.902 12.552 12.552 0 012.117 6.978 12.553 12.553 0 01-3.676 8.892 12.558 12.558 0 01-8.89 3.681l.01-.013z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M137.323 100.545c-3.956-5.443-9.71-7.973-15.713-7.973 0 0-14.703 0-19.057 13.163-1.606 4.85-1.25 10.117.697 14.84a27.046 27.046 0 002.747 5.107c3.796 5.483 15.613 20.46 15.613 20.46s13.667-18.78 16.667-24.667c3.11-6.143 3.59-14.68-.954-20.93zm-15.83 20.404a8.61 8.61 0 01-8.454-10.297 8.614 8.614 0 0111.75-6.28 8.61 8.61 0 015.318 7.963 8.555 8.555 0 01-8.614 8.614z"
              fill={4285637118}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M121.446 103.522a8.694 8.694 0 00-8.193 5.79 7.447 7.447 0 0112.147-2.109 7.45 7.45 0 11-11.237 9.752 8.693 8.693 0 107.283-13.433z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        align="center"
        color="gray-dark"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#4B4F58",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 52,
              "lineHeight": 78,
              "textAlign": "center",
            },
            Object {
              "marginVertical": 40,
            },
          ]
        }
        variant="h2"
      >
        Test
      </Text>
      <Text
        align="center"
        color="gray"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#949494",
              "fontFamily": "OpenSans_400Regular",
              "fontSize": 28,
              "lineHeight": 42,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="body1"
      >
        testing
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`IllustrationPageTemplate Without description 1`] = `
<RCTSafeAreaView
  color="white"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingBottom": 48,
          "paddingLeft": 48,
          "paddingRight": 48,
          "paddingTop": 48,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "height": undefined,
            "width": 450,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M67.4 26.49s-33.146 16.903-40.16 54c-7.013 37.097 12.82 74.6 28.044 105.333 15.223 30.734 46.446 45.2 75.496 35.917 29.05-9.283 23.724-35.74 47.307-62.56 23.583-26.82 34.333-30.483 38.043-65.483 3.71-35-31.746-70.334-76.666-76.487-44.92-6.153-72.064 9.28-72.064 9.28z"
              fill={4281240031}
              opacity={0.08}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M107.363 128.572a1.408 1.408 0 01-.87 1.308 1.419 1.419 0 01-1.541-.304 1.42 1.42 0 01-.308-1.541c.107-.258.288-.479.52-.635a1.418 1.418 0 011.784.175c.265.264.414.623.415.997z"
              fill={4292927712}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M139.856 160.388a44.426 44.426 0 01-63.776-37.293 44.424 44.424 0 0111.108-32.142 44.433 44.433 0 0177.595 26.817 44.426 44.426 0 01-11.113 32.142 44.431 44.431 0 01-13.814 10.476z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeMiterlimit",
                ]
              }
              stroke={4280693304}
              strokeMiterlimit={10}
              strokeWidth={0.5}
            />
            <RNSVGPath
              d="M134.747 100.275c-3.957-5.444-9.71-7.98-15.714-7.98 0 0-14.703 0-19.056 13.16-1.604 4.85-1.25 10.12.696 14.843a27.235 27.235 0 002.75 5.107c3.794 5.483 15.61 20.456 15.61 20.456h2.57c3.297-4.58 11.6-19.796 14.087-24.666 3.12-6.134 3.6-14.67-.943-20.92zm-13.26 24.333a12.567 12.567 0 01-11.607-7.751 12.563 12.563 0 012.721-13.689 12.553 12.553 0 0119.325 1.902 12.552 12.552 0 012.117 6.978 12.553 12.553 0 01-3.676 8.892 12.558 12.558 0 01-8.89 3.681l.01-.013z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M137.323 100.545c-3.956-5.443-9.71-7.973-15.713-7.973 0 0-14.703 0-19.057 13.163-1.606 4.85-1.25 10.117.697 14.84a27.046 27.046 0 002.747 5.107c3.796 5.483 15.613 20.46 15.613 20.46s13.667-18.78 16.667-24.667c3.11-6.143 3.59-14.68-.954-20.93zm-15.83 20.404a8.61 8.61 0 01-8.454-10.297 8.614 8.614 0 0111.75-6.28 8.61 8.61 0 015.318 7.963 8.555 8.555 0 01-8.614 8.614z"
              fill={4285637118}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M121.446 103.522a8.694 8.694 0 00-8.193 5.79 7.447 7.447 0 0112.147-2.109 7.45 7.45 0 11-11.237 9.752 8.693 8.693 0 107.283-13.433z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        align="center"
        color="gray-dark"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#4B4F58",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 52,
              "lineHeight": 78,
              "textAlign": "center",
            },
            Object {
              "marginVertical": 40,
            },
          ]
        }
        variant="h2"
      >
        Test
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E89DF",
          "borderRadius": 16,
          "justifyContent": "center",
          "minHeight": 54,
          "opacity": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
          "width": "100%",
        }
      }
    >
      <Text
        align="center"
        color="white"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 36,
              "lineHeight": 54,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="h3"
      >
        Test
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`IllustrationPageTemplate Without illustration 1`] = `
<RCTSafeAreaView
  color="white"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingBottom": 48,
          "paddingLeft": 48,
          "paddingRight": 48,
          "paddingTop": 48,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        align="center"
        color="gray-dark"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#4B4F58",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 52,
              "lineHeight": 78,
              "textAlign": "center",
            },
            Object {
              "marginVertical": 40,
            },
          ]
        }
        variant="h2"
      >
        Test
      </Text>
      <Text
        align="center"
        color="gray"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#949494",
              "fontFamily": "OpenSans_400Regular",
              "fontSize": 28,
              "lineHeight": 42,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="body1"
      >
        testing
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E89DF",
          "borderRadius": 16,
          "justifyContent": "center",
          "minHeight": 54,
          "opacity": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
          "width": "100%",
        }
      }
    >
      <Text
        align="center"
        color="white"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 36,
              "lineHeight": 54,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="h3"
      >
        Test
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`IllustrationPageTemplate Without title 1`] = `
<RCTSafeAreaView
  color="white"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingBottom": 48,
          "paddingLeft": 48,
          "paddingRight": 48,
          "paddingTop": 48,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "height": undefined,
            "width": 450,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M67.4 26.49s-33.146 16.903-40.16 54c-7.013 37.097 12.82 74.6 28.044 105.333 15.223 30.734 46.446 45.2 75.496 35.917 29.05-9.283 23.724-35.74 47.307-62.56 23.583-26.82 34.333-30.483 38.043-65.483 3.71-35-31.746-70.334-76.666-76.487-44.92-6.153-72.064 9.28-72.064 9.28z"
              fill={4281240031}
              opacity={0.08}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M107.363 128.572a1.408 1.408 0 01-.87 1.308 1.419 1.419 0 01-1.541-.304 1.42 1.42 0 01-.308-1.541c.107-.258.288-.479.52-.635a1.418 1.418 0 011.784.175c.265.264.414.623.415.997z"
              fill={4292927712}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M139.856 160.388a44.426 44.426 0 01-63.776-37.293 44.424 44.424 0 0111.108-32.142 44.433 44.433 0 0177.595 26.817 44.426 44.426 0 01-11.113 32.142 44.431 44.431 0 01-13.814 10.476z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeMiterlimit",
                ]
              }
              stroke={4280693304}
              strokeMiterlimit={10}
              strokeWidth={0.5}
            />
            <RNSVGPath
              d="M134.747 100.275c-3.957-5.444-9.71-7.98-15.714-7.98 0 0-14.703 0-19.056 13.16-1.604 4.85-1.25 10.12.696 14.843a27.235 27.235 0 002.75 5.107c3.794 5.483 15.61 20.456 15.61 20.456h2.57c3.297-4.58 11.6-19.796 14.087-24.666 3.12-6.134 3.6-14.67-.943-20.92zm-13.26 24.333a12.567 12.567 0 01-11.607-7.751 12.563 12.563 0 012.721-13.689 12.553 12.553 0 0119.325 1.902 12.552 12.552 0 012.117 6.978 12.553 12.553 0 01-3.676 8.892 12.558 12.558 0 01-8.89 3.681l.01-.013z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M137.323 100.545c-3.956-5.443-9.71-7.973-15.713-7.973 0 0-14.703 0-19.057 13.163-1.606 4.85-1.25 10.117.697 14.84a27.046 27.046 0 002.747 5.107c3.796 5.483 15.613 20.46 15.613 20.46s13.667-18.78 16.667-24.667c3.11-6.143 3.59-14.68-.954-20.93zm-15.83 20.404a8.61 8.61 0 01-8.454-10.297 8.614 8.614 0 0111.75-6.28 8.61 8.61 0 015.318 7.963 8.555 8.555 0 01-8.614 8.614z"
              fill={4285637118}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M121.446 103.522a8.694 8.694 0 00-8.193 5.79 7.447 7.447 0 0112.147-2.109 7.45 7.45 0 11-11.237 9.752 8.693 8.693 0 107.283-13.433z"
              fill={4284447739}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        align="center"
        color="gray"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#949494",
              "fontFamily": "OpenSans_400Regular",
              "fontSize": 28,
              "lineHeight": 42,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="body1"
      >
        testing
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E89DF",
          "borderRadius": 16,
          "justifyContent": "center",
          "minHeight": 54,
          "opacity": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
          "width": "100%",
        }
      }
    >
      <Text
        align="center"
        color="white"
        paragraph={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans_700Bold",
              "fontSize": 36,
              "lineHeight": 54,
              "textAlign": "center",
            },
            Object {},
          ]
        }
        variant="h3"
      >
        Test
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
